packages/@lwc/template-compiler/src/shared/ir.ts

```diff
@@ -106,5 +105,8 @@ export function isBoundToIterator(identifier: TemplateIdentifier, node?: IRNode)
     }
 
     // Delegate to parent component if no binding is found at this point
-    return isBoundToIterator(identifier, node.parent);
+    return isBoundToIterator(identifier, irParentMap.get(node));
 }
+
+export const irParentMap: WeakMap<IRElement | IRText, IRElement> = new WeakMap();
+export const irElementMap: WeakMap<IRElement, HTMLElement> = new WeakMap();
```

Comment:
Based on the numbers of occurrences of `irElementMap` references in the code and also because of the previous comment. We should probably move back the `__original` ref to the `IRElement`. Theoretically, we should not keep a reference to `__original` at all, because it means that the `IRElement` is not sufficient.